app/controllers/answers_controller.rb

```diff
@@ -1,62 +1,76 @@
 class AnswersController < ApplicationController
-  before_action :answer # , only: [:show, :edit, :update, :destroy]
-  before_action :question #, only: [:show, :edit, :update, :destroy]
+  before_action :set_answer, only: [:show, :edit, :update, :like, :dislike, :undislike, :unlike]
+  before_action :question, only: [:new, :show, :edit, :update, :destroy]
 
   def new
-    # @answer = Answer.new
-    @answer.question_id = @question.id
+  end
+
+  def show
+  end
+
+  def edit
+  end
+
+  def update
+    respond_to do |format|
+      if @answer.update(answer_params)
+        format.html { redirect_to @question }
+        format.json { render json: @question}
+      else
+        format.html { render :edit }
+        format.json { render json: @answer.errors, status: :unprocessable_entity }
+      end
+    end
   end
 
   def create
     @answer = Answer.new(answer_params)
     @answer.question_id = params[:question_id]
+    @answer.user = current_user
     @answer.save
     redirect_to question_path(@answer.question)
   end
 
+
+  def destroy
+    @answer = Answer.find(params[:id])
+    @answer.destroy
+   
+    redirect_to :back
+  end
+
   def like
-    @answer.liked_by @user
+    @answer.liked_by current_user
```

Comment:
Yay! Thank you! Have I mentioned how much I LOVE `current_user`? It's literally the best thing ever.